### PR TITLE
Feature/add profiling support

### PR DESCRIPTION
**Pull request by dutsuwak**

### CODE REVIEW

File: meson_options.txt [hunk 1]
```diff
@@ -2,6 +2,7 @@
 option('enable-tests', type : 'feature', value : 'auto', yield : true, description : 'Build tests')
 option('enable-examples', type : 'feature', value : 'auto', yield : true, description : 'Build examples')
 option('enable-gtk-doc', type : 'boolean', value : false, description : 'Use gtk-doc to build documentation')
+option('profiling', type : 'feature', value : 'disabled', yield : true, description: 'Enable profiling building')
```
Comment (mleivaf) [inline meson_options.txt]: Could we change the parameter name to `enable-profiling`?

File: meson.build [hunk 2]
```diff
@@ -33,6 +33,19 @@ example_deps = [gst_dep, gst_base_dep]
...
+# Verify if profiling was enabled
+if get_option('profiling').enabled()
+  profiler_dep = dependency('libprofiler')
+  if(profiler_dep.found())
+    message('Profiling enabled: Building examples with profiling support.')
+    link_flags = ['-lprofiler']
+    # Update test test_deps to include profiler dependency
+    test_deps=[gst_dep, gst_base_dep, gst_check_dep, profiler_dep]
+  else
+    error('MESON_SKIP gperftools profiling library not found.')
```
Comment (michaelgruner) [inline meson.build]: If the user enabled it, we should fail instead of skipping.

**Review by michaelgruner — CHANGES_REQUESTED** [review]

File: meson.build [hunk 3]
```diff
@@ -33,6 +33,19 @@ example_deps = [gst_dep, gst_base_dep]
...
 test_deps=[gst_dep, gst_base_dep, gst_check_dep]
 
+# Verify if profiling was enabled
+if get_option('profiling').enabled()
+  profiler_dep = dependency('libprofiler')
+  if(profiler_dep.found())
+    message('Profiling enabled: Building examples with profiling support.')
+    link_flags = ['-lprofiler']
+    # Update test test_deps to include profiler dependency
+    test_deps=[gst_dep, gst_base_dep, gst_check_dep, profiler_dep]
```
Comment (mleivaf) [inline meson.build]: Can we add only the new dependency to the list of default dependencies?
For example
`test_deps+=profiler_dep`

**Review by mleivaf — CHANGES_REQUESTED** [review]

**Review by mleivaf — CHANGES_REQUESTED** [review]
Can we create a special definition for profiling? in order to use this definition to enable conditional code.

**Review by michaelgruner — APPROVED** [review]

**Pull request merged**